src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBaseKernels.hpp

```diff
@@ -344,12 +342,12 @@ class PhaseVolumeFractionKernel : public PropertyKernelBase< NUM_COMP >
       phaseVolFrac[ip] = phaseFrac[ip] * phaseDensInv;
 
       dPhaseVolFrac_dPres[ip] =
-        (dPhaseFrac_dPres[ip] - phaseVolFrac[ip] * dPhaseDens_dPres[ip]) * phaseDensInv;
+        (dPhaseFrac[ip][Deriv::dP] - phaseVolFrac[ip] * dPhaseDens[ip][Deriv::dP]) * phaseDensInv;
 
       for( integer jc = 0; jc < numComp; ++jc )
       {
         dPhaseVolFrac_dComp[ip][jc] =
-          (dPhaseFrac_dComp[ip][jc] - phaseVolFrac[ip] * dPhaseDens_dComp[ip][jc]) * phaseDensInv;
+          (dPhaseFrac[ip][Deriv::dC+jc] - phaseVolFrac[ip] * dPhaseDens[ip][Deriv::dC+jc]) * phaseDensInv;
```

Comment:
I bet this `Deriv::dC+jc` is a trick to put keep the same dimension for all your arrays?

Reply:
Yes, in the current version of the PR, the derivatives are grouped and ordered as follows:

- 0: derivative wrt pressure
- 1: derivative wrt temperature
- 2: first derivative wrt component fraction

With this approach, there is only one array storing derivatives for a given physical quantity. For instance, for the phase component fractions (of size: `numPhases x numComps`), there is one array of derivatives of size `numPhases x numComps x numDofs`, with `numDofs = numComps + 2`.